Source/WebKit/UIProcess/Extensions/WebExtension.cpp

```diff
@@ -2220,6 +2227,124 @@ void WebExtension::populateCommandsIfNeeded()
     }
 }
 
+std::optional<WebExtension::DeclarativeNetRequestRulesetData> WebExtension::parseDeclarativeNetRequestRulesetObject(const JSON::Object& rulesetObject, RefPtr<API::Error>& error)
+{
+    auto rulesetID = rulesetObject.getString(declarativeNetRequestRulesetIDManifestKey);
+    if (rulesetID.isEmpty()) {
+        error = createError(Error::InvalidDeclarativeNetRequest, WEB_UI_STRING("Empty `declarative_net_request` ruleset id.", "WKWebExtensionErrorInvalidDeclarativeNetRequestEntry description for empty ruleset id"));
+        return { };
+    }
+
+    auto jsonPath = rulesetObject.getString(declarativeNetRequestRulePathManifestKey);
+    if (jsonPath.isEmpty()) {
+        error = createError(WebExtension::Error::InvalidDeclarativeNetRequest, WEB_UI_STRING("Empty `declarative_net_request` JSON path.", "WKWebExtensionErrorInvalidDeclarativeNetRequestEntry description for empty JSON path"));
+        return { };
+    }
+
+    auto enabledBool = rulesetObject.getBoolean(declarativeNetRequestRuleEnabledManifestKey);
+    if (!enabledBool) {
+        error = createError(WebExtension::Error::InvalidDeclarativeNetRequest, "`declarative_net_request` missing enabled boolean"_s);
```

Comment:
```suggestion
        error = createError(WebExtension::Error::InvalidDeclarativeNetRequest, WEB_UI_STRING("Missing or invalid `enabled` boolean for the `declarative_net_request` manifest entry.", "WKWebExtensionErrorInvalidDeclarativeNetRequestEntry description for missing enabled boolean"));
```

And re-run `update-webkit-localizable-strings` to update the strings from the source code.